### PR TITLE
Update RazorPage template to be able to be SxS with previous version

### PR DESCRIPTION
**Pull request by phenning**
- Update Web.ItemTemplates to use a versioned package identity
- Bump up the precedence
- Make unique identities
- Add Framework symbol for RazorPage template

This fixes the issue with template creation in Visual Studio where the NET 6 template hides the NET 5 version of the template resulting in errors due to the file-scoped namespace usage in the net 6 version of the template.

### CODE REVIEW

**Comment by phenning** [comment]
@mkArtakMSFT  is there anything else that needs to be updated for this rename? I already have a pending PR to the installer.

**Comment by mkArtakMSFT** [comment]
> @mkArtakMSFT is there anything else that needs to be updated for this rename? I already have a pending PR to the installer.

I think you've covered everything.  @javiercn can you also please look at this? Thanks!

**Comment by phenning** [comment]
Note that this solves the issue for the Visual Studio scenarios (along with some associated changes in VS), however, this does not solve the problem for where a user has NET 5 and NET 6 installed and does "dotnet new razorpage" on a NET 5 project, since this template would still win due to being the highest precedence.

We have a number of options here:
1. Have this template support "net5.0" "netcoreapp3.1", etc and condition the namespace based on that.

2. Have a "UseScopedNamespace" or similar parameter with a default of true. Customers wanting to create templates for NET 5 or earlier would need to specify false. 

3. Along with the other options, we could potentially have VS implement a host property that the template could bind to the LangVersion of the project. This would give us some flexibility on whether to apply the new namespace syntax based on the detected version. Obviously the dotnet new would not provide this property, so any conditions would need to account for that.

I think the simplest lowest risk solution would be #2, but VS may end up supporting these host properties anyway, but we wouldn't need to use it.

**Comment by DamianEdwards** [comment]
@phenning I think we should go with option 1 (and potentially option 3) to ensure that this template keeps playing well with earlier project types when used from the CLI, without the user having to specify a parameter. Could we just have it use "UseFileScopedNamespace" parameter which defaults to "true" only if the TFM is net6.0+?

**Comment by javiercn** [comment]
@phenning @DamianEdwards is there anything blocking this PR?

**Comment by phenning** [comment]
@javiercn  there will be another change incoming.

**Comment by DamianEdwards** [comment]
@phenning note that as per ongoing email thread RE file-spaced namespace defaults and editorconfig, we should **not** have any template options related to language version specific features. The IDE will auto fix-up the namespace declaration on file creation to match our desired defaults and user settings in editorconfig.

**Comment by phenning** [comment]
@DamianEdwards  this was the list of changes I took last week. Is this still valid?

* Remove "Fab 4" using statments
* Move back to Block namespace
* Remove framework symbol from template.json
* Keep all the other authoring changes as relating to template identity and package identity

**Comment by DamianEdwards** [comment]
@phenning I think the first item is wrong, the using statements should only be removed if the MSBuild property `ImplicitUsings` is enabled and we don't have that info in the template yet, so we're going to keep the usings there to ensure the template works in all projects. The others look right.

**Comment by phenning** [comment]
@DamianEdwards I don't mean remove ALL the usings, just the ones below, since they aren't directly used by the template (this is the change we just made to the VS templates as well)

If we keep the using statements in this template, do we want to do that as well in the Visual Studio templates? Late last week we had said that we would remove any using statements that a particular item template did not explicitly use. If so, we can revert that change to the Visual Studio templates and just put the usings all back in - essentially we'll be reverting all the changes we made to those templates thus far.

REMOVE:
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

KEEP:
using Microsoft.AspNetCore.Mvc;
using Microsoft.AspNetCore.Mvc.RazorPages;

**Comment by DamianEdwards** [comment]
@phenning ah sorry you're correct. Just keep the usings required for the file to compile.

**Comment by phenning** [comment]
@DamianEdwards Thanks!
@javiercn  I think this should be good now. After this completes, this PR will need to be completed in the installer: https://github.com/dotnet/installer/pull/11190

**Review by DamianEdwards — APPROVED** [review]

**Comment by phenning** [comment]
@javiercn what needs to be done here? I'm not super familiar with your repo our how to debug issues.

**Comment by DamianEdwards** [comment]
@phenning test failures here: https://dev.azure.com/dnceng/public/_build/results?buildId=1284191&view=ms.vss-test-web.build-test-results-tab&runId=37917650&resultId=100610&paneView=debug

I kicked off that build again in case it was transitory

**Comment by phenning** [comment]
> @phenning test failures here: https://dev.azure.com/dnceng/public/_build/results?buildId=1284191&view=ms.vss-test-web.build-test-results-tab&runId=37917650&resultId=100610&paneView=debug
> 
> I kicked off that build again in case it was transitory

Thanks, I have zero context here on what is being tested or how these relatively small content changes could impact these tests. Unless there is an issue with the package rename. If that's the case, some assistance would be appreciated.

**Comment by DamianEdwards** [comment]
@javiercn @pranavkm can someone help @phenning diagnose the issue here please?

**Comment by phenning** [comment]
@DamianEdwards, @mkArtakMSFT 

It looks like this never made it in before the branch, I'll leave it up to you and aspnet devs to get this into NET 6 and beyond, since I don't know what is broken here in the repo.  Feel free to abandon this one and just pull in the namespace changes.

**Comment by javiercn** [comment]
This still goes to Main and then we'll back-port it to the appropriate release branch

**Comment by DamianEdwards** [comment]
@javiercn are you able to pick this one up and get it in?

**Comment by javiercn** [comment]
@DamianEdwards I won't get to it until Monday. (It's friday afternoon here).

I believe the issues we are seeing on CI are known (@SteveSandersonMS knows the details).

**Comment by javiercn** [comment]
@pranavkm we merged the original backport into 6.0 but we didn't merge this, am I wrong here?

**Comment by pranavkm** [comment]
Sorry I just saw @javiercn's comment now. It doesn't look like this PR made it's way to 6.0 or main. @phenning I imagine we want this change ported to 6.0, is that right?

**Comment by phenning** [comment]
> Sorry I just saw @javiercn's comment now. It doesn't look like this PR made it's way to 6.0 or main. @phenning I imagine we want this change ported to 6.0, is that right?

Yes especially if this is the change to the package identity, failure to take that could make, otherwise there could be some real issues with template resolution later.

**Comment by pranavkm** [comment]
Thanks. Let me see if we can schedule this for 6.0.1. The change went in to 6.0-rc1, but we missed getting it in to 6.0-rc2 or later.

**Comment by phenning** [comment]
That would be great. I’m really not sure of the behavior this will have in regards to resolution of templates in Visual Studio with mismatched template package identifies between RC1 and GA then servicing.

**Pull request merged**
